media/js/newsletter/management.es6.js

```diff
@@ -0,0 +1,785 @@
+ * @param {String} url
+ * @param {String} method
+ * @returns {Promise}
+ */
+function _fetch(url, method) {
+    return new window.Promise((resolve, reject) => {
+        window
+            .fetch(url, {
+                method: method,
+                headers: {
+                    Accept: 'application/json',
+                    'Content-Type': 'application/json'
+                }
+            })
+            .then((resp) => {
+                if (
+                    resp.statusText === 'OK' &&
+                    resp.status >= 200 &&
+                    resp.status < 300
+                ) {
+                    resolve(resp.json());
+                } else {
+                    reject(resp);
+                }
+            })
+            .catch((e) => {
+                reject(e);
+            });
+    });
+}
+
+/**
+ * Main newsletter management form object.
+ */
+const NewsletterManagementForm = {
+    meetsRequirements: () => {
+        return 'Promise' in window && 'fetch' in window;
+    },
+
+    getPageLocale: (locale) => {
+        return typeof locale !== 'undefined'
+            ? locale
+            : document.getElementsByTagName('html')[0].getAttribute('lang');
+    },
+
+    getPageURL: () => {
+        return window.location.href;
+    },
+
+    getUserEmail: () => {
+        return _userData ? _userData.email : null;
+    },
+
+    getFormCountry: () => {
+        return _form.querySelector('select[name="country"]').value;
+    },
+
+    getFormLang: () => {
+        return _form.querySelector('#id_lang').value;
+    },
+
+    /**
+     * Get an array of checked newsletter IDs from the management form.
+     * @returns {Array}
+     */
+    getCheckedNewsletters: () => {
+        return Array.from(
+            document.querySelectorAll(
+                '.newsletter-table .newsletter-checkbox:checked'
+            )
+        ).map((newsletter) => {
+            return `${newsletter.value}`;
+        });
+    },
+
+    /**
+     * Get Basket URL for querying user data.
+     * @returns {String}
+     */
+    getUserDataURL: () => {
+        return _form.getAttribute('action');
+    },
+
+    /**
+     * Get Basket URL for querying newsletter data.
+     * @returns {String}
+     */
+    getNewsletterDataURL: () => {
+        return _form.getAttribute('data-newsletters-url');
+    },
+
+    /**
+     * Get Bedrock URL for querying localised newsletter strings.
+     * @returns {String}
+     */
+    getNewsletterStringsURL: () => {
+        return _form.getAttribute('data-strings-url');
+    },
+
+    /**
+     * Is the given locale in the list of supported locales for Firefox Account newsletters?
+     * @param {String} locale
+     * @returns  {Boolean}
+     */
+    isFxALocale: (locale) => {
+        const loc = locale.includes('-') ? locale.split('-')[0] : locale;
+
+        return FXA_NEWSLETTERS_LOCALES.reduce((prev, current) => {
+            return loc === current ? true : prev;
+        }, false);
```

Comment:
Would `FXA_NEWSLETTERS_LOCALES.includes(loc)` work? Or is it not well supported enough?

Reply:
Heh, yeah I even used `includes()` in the function right underneath. Not sure why I ended up with this at some point :)